### PR TITLE
Application crd prototype

### PR DESCRIPTION
**Pull request by irfanurrehman**
This discovers `applications cr` with its selected entities (kubernetes resources) and creates dtos for it as a `business application`. 
~~The relationship with selected resources is the next step (There are some confusions).~~
~~This can either be hidden behind a flag or submitted to a feature branch.~~
~~The unit test might fail, as I haven't updated the tests yet.~~
EDIT: 
This relationship with selected resources is now complete.
No flag is needed.
The unit tests are fixed.

### CODE REVIEW

File: pkg/registration/supply_chain_factory.go [hunk 1]
```diff
@@ -177,8 +177,15 @@ func (f *SupplyChainFactory) createSupplyChain() ([]*proto.TemplateDTO, error) {
...
 
+	businessAppSupplyChainNode, err := f.buildBusinessApplicationSupplyBuilder()
+	if err != nil {
+		return nil, err
+	}
+	glog.V(4).Infof("Supply chain node: %+v", businessAppSupplyChainNode)
+
 	supplyChainBuilder := supplychain.NewSupplyChainBuilder()
 	supplyChainBuilder.Top(serviceSupplyChainNode)
+	supplyChainBuilder.Entity(businessAppSupplyChainNode)
```
Comment (enlinxu) [inline pkg/registration/supply_chain_factory.go:188]: businessapp should be the Top

**Review by enlinxu — APPROVED** [review]
LGTM except the supply chain definition

File: pkg/registration/supply_chain_factory.go (hunk 1 above)
Comment (irfanurrehman) [reply pkg/registration/supply_chain_factory.go]: Thanks for the review @enlinxu. Updated, PTAL!

**Comment by irfanurrehman** [comment]
Rebased!

**Review by esara — APPROVED** [review]

**Comment by irfanurrehman** [comment]
@enlinxu Should I merge this?

**Comment by enlinxu** [comment]
> @enlinxu Should I merge this?

8.0.3 is a big release and I want to control any possible risk. Let's merge this as soon as after 8.0.3 release is created (next Monday)

**Pull request merged**